### PR TITLE
Feature/pratyush1718/bookmark component

### LINKED ISSUE #15: Bookmarks
Create a file with functions that allow creating new bookmarks, deleting existing ones, and fetching all bookmarks. These should be stored on device's local storage. The title of the page being bookmarked should be sufficient.

### PR DESCRIPTION
**Pull request by pratyush1718**
## Tracking Info

Resolves #15 

## Changes

Added AsyncStorage functionality to store names of bookmarked pages, added a bookmark component to emergency and general principles page to leverage said functionality

## Testing

Added console log statements for bookmark routes and tested component on android emulator

## Confirmation of Change

image
image

### CODE REVIEW

File: dfm-sideline-sidekick-app/components/bookmarkRoutes.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  try {
+    const existingBookmarks = JSON.parse(await AsyncStorage.getItem('bookmarks')) || [];
+    return existingBookmarks;
+  } catch (error) {
+    console.error('Error getting bookmarks:', error);
+    return [];
+  }
+}
+
+// Function for testing to clear all bookmarks from AsyncStorage
```
Comment (Anthonyp0329) [inline dfm-sideline-sidekick-app/components/bookmarkRoutes.js]: can remove this code for merging

**Review by Anthonyp0329 — CHANGES_REQUESTED** [review]
Looking awesome, great work! Just make sure to add this to the ConditionsSection.tsx page as well.

**Review by Anthonyp0329 — APPROVED** [review]

**Review by r800360 — APPROVED** [review]
I went ahead and changed the bookmarkRoutes.js to a .tsx file and created the Bookmark type to fix all the linter issues that were popping up with that file. I also switched TouchableOpacity to the industry standard Pressable in the bookmark.tsx. These changes look good to me and work on my end, but be sure to check that everything works as expected and fix any unresolved issues that Anthony mentioned. Nice work on this!

**Review by EdwardLinS — APPROVED** [review]
Good job Pratyush! It looks good to me. Tested different bookmarks on iPad and Android devices and they all seem to work. Other than the things Anthony said, seems ready to merge.

**Pull request merged**